standalone/runtime/tests/test_vault_registry.rs

```diff
@@ -343,3 +363,144 @@ fn integration_test_vault_registry_register_respects_fund_limit() {
+                vault.to_be_replaced = vault_id.wrapped(0);
+                vault.backing_collateral = Amount::new(0, currency_id);
+                vault.liquidated_collateral = Amount::new(0, currency_id);
+                *vault
+                    .free_balance
+                    .get_mut(&vault_data.replace_collateral.currency())
+                    .unwrap() += vault_data.replace_collateral;
+            })
+        );
+        deposit_collateral_and_issue(vault_id);
+    });
+}
+
+#[test]
+fn integration_test_vault_registry_theft_recovery_fails() {
+    test_with(|vault_id| {
+        liquidate_vault_with_status(&vault_id, VaultStatus::CommittedTheft);
+        // `to_be_redeemeded` tokens are non-zero
+        assert_err!(
+            Call::VaultRegistry(VaultRegistryCall::recover_vault_id {
+                currency_pair: vault_id.currencies.clone(),
+            })
+            .dispatch(origin_of(account_of(VAULT))),
+            VaultRegistryError::VaultNotRecoverable
+        );
+    });
+}
+
+#[test]
+fn integration_test_vault_registry_theft_recovery_works() {
+    test_with(|vault_id| {
+        let currency_id = vault_id.collateral_currency();
+        let mut vault_data = default_vault_state(&vault_id);
+        vault_data.to_be_redeemed = vault_id.wrapped(0);
+        vault_data.to_be_replaced = vault_id.wrapped(0);
+        CoreVaultData::force_to(&vault_id, vault_data.clone());
+
+        liquidate_vault_with_status(&vault_id, VaultStatus::CommittedTheft);
+
+        assert_ok!(Call::VaultRegistry(VaultRegistryCall::recover_vault_id {
+            currency_pair: vault_id.currencies.clone(),
+        })
+        .dispatch(origin_of(account_of(VAULT))));
+
+        assert_eq!(
+            ParachainState::get(&vault_id),
+            ParachainState::get_default(&vault_id).with_changes(|_, vault, liquidation_vault, _| {
+                let liquidation_vault = liquidation_vault.with_currency(&vault_id.currencies);
+
+                liquidation_vault.collateral =
+                    Amount::new(default_vault_backing_collateral(currency_id).amount(), currency_id);
+                liquidation_vault.to_be_issued = vault_id.wrapped(DEFAULT_VAULT_TO_BE_ISSUED.amount());
+                liquidation_vault.issued = vault_id.wrapped(DEFAULT_VAULT_ISSUED.amount());
+                liquidation_vault.to_be_redeemed = vault_id.wrapped(0);
+
+                vault.issued = vault_id.wrapped(0);
+                vault.to_be_issued = vault_id.wrapped(0);
+                vault.to_be_redeemed = vault_id.wrapped(0);
+                vault.to_be_replaced = vault_id.wrapped(0);
+                vault.backing_collateral = Amount::new(0, currency_id);
+                vault.liquidated_collateral = Amount::new(0, currency_id);
+                *vault
+                    .free_balance
+                    .get_mut(&vault_data.replace_collateral.currency())
+                    .unwrap() += vault_data.replace_collateral;
```

Comment:
Why did the vault not receive the collateral that was above [the liquidation threshold at the time of liquidation]?

Reply:
That's because in this setup the vault was undercollateralized. I realised this test wasn't testing changes caused by the actual recovery extrinsic, so I modified it slightly.